### PR TITLE
fix smartlock

### PR DESCRIPTION
**Pull request by thieren**
In the current version the raw `DeviceLockState` property was not converted properly (would be `true/false` instead of `3, 4, 5`. This leads to the Lock always reported as Unlocked [here](https://github.com/bropat/eufy-security-client/blob/54f94083592feeeaf8e53ab77b521f4cd62aea04/src/http/device.ts#L1841).

Also if the lock is manually locked/unlocked the `DeviceLocked` property was not updated correctly.

### CODE REVIEW

**Comment by bropat** [comment]
Function `convertRawPropertyValue` was removed from class `Lock` in branch `develop`.

**Comment by thieren** [comment]
I see. This just leaves the update to `PropertyName.DeviceLocked` when lock is changed manually.

**Comment by bropat** [comment]
Please look into comments above. (review)

**Comment by thieren** [comment]
I'm not sure that I am following you here. The only comment I am seeing is that `convertRawPropertyValue`is removed, since the method of `Device` is essentially doing the same. But that still leaves the updates to `PropertyName.DeviceLocked` - am I not right?

**Comment by bropat** [comment]
There are 4 more comments before the last one under the review...

image

**Comment by thieren** [comment]
I don't see any review tab or similar...

But since I can read your comment in the screenshot:
I will test again...

Bildschirmfoto 2022-07-21 um 05 49 38

**Comment by bropat** [comment]
I removed also some unneeded emitting of "locked" events. Please try the latest develop branch and let me know.

Thanks!

**Comment by jaredallard** [comment]
@bropat I think you forgot to submit the review, since the screnshots show it as "Pending" :D (thx Github!)

**Comment by thieren** [comment]
> I removed also some unneeded emitting of "locked" events. Please try the latest develop branch and let me know.
> 
> Thanks!

Hi @bropat 

if these emitted events were needed, depends somehow on the way you want it to work I guess. I'll use our homebridge plugin as example:

If the user changed the state of the lock, we'll send the appropriate command to the client. Homebridge sets the target state of the log (e.g. "Locked" as Target; CurrentState will remain as "Unlocked").
The Plugin then waits for the state change... By now this is handled via the event "locked" from the device. Since this would just be emitted when the lock is truly locked (not just the command).
Now that the event is no longer emitted we'd have to fall back on the emitted property change (What we'll probably do anyway, since the "locked" event didn't support the "MECHANICAL FAILURE/JAMMED" state.)

So in short, I'd favour to get the "locked" event back, and if just for backwards compatibility and a little more clearer states. But really 'needed' are they not.

As for the updates to `PropertyName.DeviceLocked`: you're right. We tested and these seem to work, without the extra update.

File: src/http/device.ts [hunk 1]
```diff
@@ -1930,6 +1933,7 @@ export class Lock extends Device {
                     {
                         const cmdType = this.isLockBle() || this.isLockBleNoFinger() ? CommandType.CMD_DOORLOCK_GET_STATE : CommandType.CMD_SMARTLOCK_QUERY_STATUS;
                         this.updateRawProperty(cmdType, "3");
+                        this.updateProperty(PropertyName.DeviceLocked, false);
```
[the remark this answers is not in the document]
Comment (thieren) [reply src/http/device.ts:1918]: You're right. Not needed.
```suggestion

```

File: src/http/device.ts [hunk 2]
```diff
@@ -1939,6 +1943,7 @@ export class Lock extends Device {
                     case LockPushEvent.MULTIPLE_ERRORS:
                     {
                         const cmdType = this.isLockBle() || this.isLockBleNoFinger() ? CommandType.CMD_DOORLOCK_GET_STATE : CommandType.CMD_SMARTLOCK_QUERY_STATUS;
+                        this.updateProperty(PropertyName.DeviceLocked, false);
```
[the remark this answers is not in the document]
Comment (thieren) [reply src/http/device.ts:1928]: You're right. Not needed.
```suggestion

```

File: src/http/device.ts [hunk 3]
```diff
@@ -1871,12 +1871,14 @@ export class Lock extends Device {
 
     protected convertRawPropertyValue(property: PropertyMetadataAny, value: string): PropertyValue {
```
[the remark this answers is not in the document]
Comment (thieren) [reply src/http/device.ts]: You're right. Not needed.
```suggestion

```

File: src/http/device.ts [hunk 4]
```diff
@@ -1918,6 +1920,7 @@ export class Lock extends Device {
                     {
                         const cmdType = this.isLockBle() || this.isLockBleNoFinger() ? CommandType.CMD_DOORLOCK_GET_STATE : CommandType.CMD_SMARTLOCK_QUERY_STATUS;
                         this.updateRawProperty(cmdType, "4");
+                        this.updateProperty(PropertyName.DeviceLocked, true);
```
[the remark this answers is not in the document]
Comment (thieren) [reply src/http/device.ts:1905]: You're right. Not needed.
```suggestion

```

**Comment by bropat** [comment]
Closed, since nothing to merge.
